### PR TITLE
Christian/button styling

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: e2e/example.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { test, expect } from '@playwright/test';
```
Comment (hein-obox) [inline e2e/example.spec.ts:1]: Please remove from this branch.

File: includes/widgets/traits/button-trait.php [hunk 2]
```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
...
+
+		$args = wp_parse_args( $args, $default_args );
+
+		$start = is_rtl() ? 'right' : 'left';
+		$end = is_rtl() ? 'left' : 'right';
+
+		$this->add_control(
+			'icon_align',
+			[
+				'label' => esc_html__( 'Icon Position', 'elementor' ),
```
Comment (hein-obox) [inline includes/widgets/traits/button-trait.php]:
```suggestion
				'label' => esc_html__( 'Position', 'elementor' ),
```

File: includes/widgets/traits/button-trait.php [hunk 3]
```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
...
+				'size_units' => [ 'px', 'em', 'rem', 'vw', 'custom' ],
+				'selectors' => [
+					'{{WRAPPER}}' => '--n-button-icon-size: {{SIZE}}{{UNIT}}',
+				],
+		] );
+
+		$this->add_control(
+			'icon_indent',
+			[
+				'label' => esc_html__( 'Icon Spacing', 'elementor' ),
```
Comment (hein-obox) [inline includes/widgets/traits/button-trait.php]:
```suggestion
				'label' => esc_html__( 'Spacing', 'elementor' ),
```

File: includes/widgets/traits/button-trait.php [hunk 4]
```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
...
+					$args['section_condition'],
+					[
+						'text!' => '',
+						'selected_icon[value]!' => '',
+					]
+				),
+			]
+		);
+
+		//start the color tabs
```
Comment (hein-obox) [inline includes/widgets/traits/button-trait.php]: Remove comment.

File: includes/widgets/traits/button-trait.php [hunk 5]
```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
...
+		$this->start_controls_tabs( 'icons_button_style', [
+			'condition' => $args['section_condition'],
+		] );
+				//when the icon is on normal
```
Comment (hein-obox) [inline includes/widgets/traits/button-trait.php:553]: Remove these types of comments.

File: includes/widgets/traits/button-trait.php [hunk 6]
```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
...
+				$this->start_controls_tab(
```
Comment (hein-obox) [inline includes/widgets/traits/button-trait.php]: Remove tabs on the left.

File: includes/widgets/traits/button-trait.php [hunk 7]
```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
...
+						'label' => esc_html__( 'Color', 'elementor' ),
+						'type' => Controls_Manager::COLOR,
+						'selectors' => [
+							'{{WRAPPER}}' => '--button-icon-color: {{VALUE}};',
+						],
+					] );
+						
+				$this->end_controls_tab();
+						
+				//when the icon is on hover
```
Comment (hein-obox) [inline includes/widgets/traits/button-trait.php]: Remove.

File: playwright.config.ts [hunk 8]
```diff
@@ -0,0 +1,77 @@
+import { defineConfig, devices } from '@playwright/test';
```
Comment (hein-obox) [inline playwright.config.ts:1]: Move to test branch.

File: tests-examples/demo-todo-app.spec.ts [hunk 9]
```diff
@@ -0,0 +1,437 @@
+import { test, expect, type Page } from '@playwright/test';
```
Comment (hein-obox) [inline tests-examples/demo-todo-app.spec.ts:1]: Move to a test branch.

File: assets/dev/scss/frontend/widgets/button.scss [hunk 10]
```diff
@@ -0,0 +1,48 @@
+.elementor-widget-button {
```
Comment (hein-obox) [inline assets/dev/scss/frontend/widgets/button.scss]: Move your styling to the existing file: `assets/dev/scss/frontend/general/_button.scss`

File: assets/dev/scss/frontend/_widgets.scss [hunk 11]
```diff
@@ -3,6 +3,7 @@
 //
 @import "widgets/accordion";
 @import "widgets/alert";
+@import "widgets/button";
```
Comment (hein-obox) [inline assets/dev/scss/frontend/_widgets.scss]: Revert this line.

File: includes/widgets/traits/button-trait.php [hunk 12]
```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
...
+					'em' => [
+						'max' => 10,
+					],
+					'rem' => [
+						'max' => 10,
+					],
+				],
+				'size_units' => [ 'px', 'em', 'rem', 'vw', 'custom' ],
+				'selectors' => [
+					'{{WRAPPER}}' => '--n-button-icon-size: {{SIZE}}{{UNIT}}',
```
Comment (hein-obox) [inline includes/widgets/traits/button-trait.php]:
```suggestion
					'{{WRAPPER}}' => '--button-icon-size: {{SIZE}}{{UNIT}}',
```

File: includes/widgets/traits/button-trait.php [hunk 13]
```diff
@@ -503,6 +438,161 @@ protected function register_button_style_controls( $args = [] ) {
...
+					] );
+						
+				$this->end_controls_tab();
+						
+				//when the icon is on hover
+				$this->start_controls_tab(
+						'icon_button_hover',
+						[
+							'label' => esc_html__( 'Hover', 'elementor' ),
+							'condition' => $args['section_condition'],
```
Comment (hein-obox) [inline includes/widgets/traits/button-trait.php]: Do we need this?
What is this adding?

File: assets/dev/scss/frontend/widgets/button.scss [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+       
+    }
+
+    .elementor-button:hover{
+
+        .elementor-button-icon {
+
+            i {
+                color: var(--button-icon-color-hover);
+                transition: all 0.3s;
```
Comment (hein-obox) [inline assets/dev/scss/frontend/widgets/button.scss]: The button hover effect has a transition duration already. Can we use that one? 
We shouldn't add a separate styling for this.

Can you please test this how the button hover transition durations works and how if we need any additional styling.

Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/button.scss]: Screenshot 2024-04-22 at 11 46 19

Comment (Mchris-hub) [reply assets/dev/scss/frontend/widgets/button.scss]: the color of the icon was loading quickly and was faster than the color of the button on hover, so I took the transition of the button and I used the same values for the icon

File: includes/widgets/traits/button-trait.php (hunk 12 above)
Comment (Mchris-hub) [reply includes/widgets/traits/button-trait.php]: i changed it

File: includes/widgets/traits/button-trait.php (hunk 3 above)
Comment (Mchris-hub) [reply includes/widgets/traits/button-trait.php]: i changed it

File: includes/widgets/traits/button-trait.php (hunk 4 above)
Comment (Mchris-hub) [reply includes/widgets/traits/button-trait.php]: i removed them

File: includes/widgets/traits/button-trait.php (hunk 13 above)
Comment (Mchris-hub) [reply includes/widgets/traits/button-trait.php]: for some reason the size does not work when I remove it

File: modules/apps/admin-apps-page.php
Comment (Mchris-hub) [inline modules/apps/admin-apps-page.php:1]: I changed for it

File: assets/dev/scss/frontend/widgets/button.scss (hunk 10 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/widgets/button.scss]: when i moved my styles to _button.scss it stops working, it seems like the styles in this file does not get applied at all

File: assets/dev/scss/frontend/_widgets.scss [hunk 15]
```diff
@@ -3,6 +3,7 @@
 //
 @import "widgets/accordion";
 @import "widgets/alert";
+// @import "widgets/button";
```
Comment (hein-obox) [inline assets/dev/scss/frontend/_widgets.scss]: Please remove.

File: assets/dev/scss/frontend/general/_button.scss [hunk 16]
```diff
@@ -25,16 +25,31 @@
...
+		}
+
 		svg {
 			width: 1em;
 			height: auto;
+			fill: var(--button-icon-color, inherit);
+		}
+
+		svg:hover {
+			fill: var(--button-icon-color, inherit);
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:46]:
```suggestion
			fill: var(--button-icon-color-hover, inherit);
```

File: assets/dev/scss/frontend/general/_button.scss [hunk 17]
```diff
@@ -25,16 +25,31 @@
...
+			color: var(--button-icon-color, inherit);
+		}
+
 		svg {
 			width: 1em;
 			height: auto;
+			fill: var(--button-icon-color, inherit);
+		}
+
+		svg:hover {
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: We will need to listen to the button hover, not for the icon hover.

`.your-wrapper-class:hover svg`

File: assets/dev/scss/frontend/general/_button.scss [hunk 18]
```diff
@@ -25,16 +25,31 @@
...
+			fill: var(--button-icon-color, inherit);
 		}
 
 		.e-font-icon-svg {
 			height: 1em;
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:44]:
```suggestion
			height: 1em;
			font-size: var(--button-icon-size);
```

File: assets/dev/scss/frontend/general/_button.scss [hunk 19]
```diff
@@ -25,16 +25,31 @@
...
 		}
 	}
 
+	.e-font-icon-svg {
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Add to the existing styling above.

File: assets/dev/scss/frontend/general/_button.scss [hunk 20]
```diff
@@ -25,16 +25,31 @@
...
+			fill: var(--button-icon-color, inherit);
+		}
+
+		svg:hover {
+			fill: var(--button-icon-color, inherit);
 		}
 
 		.e-font-icon-svg {
 			height: 1em;
 		}
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:45]:
```suggestion
		}
		
		&:hover {
			.elementor-button-icon {

				i {
					color: var(--button-icon-color-hover);
					transition: all 0.3s;
			}

			svg {
				fill: var(--button-icon-color-hover);
				transition: all 0.3s;
			}
		}
	}
		
		
```

File: assets/dev/scss/frontend/general/_button.scss [hunk 21]
```diff
@@ -83,6 +98,23 @@
 
 	span {
 		text-decoration: inherit; //fix for conflict with text-decoration & inline-block
+		align-self: center;
+	}
+}
+
+.elementor-button:hover{
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Add more up inside the nesting.

File: assets/dev/scss/frontend/widgets/button.scss [hunk 22]
```diff
@@ -0,0 +1,45 @@
+.elementor-widget-button {
```
Comment (hein-obox) [inline assets/dev/scss/frontend/widgets/button.scss]: Please delete this file.

File: e2e/example.spec.ts (hunk 1 above)
Comment (hein-obox) [reply e2e/example.spec.ts:1]: Please remove file.

File: modules/apps/admin-apps-page.php [hunk 23]
```diff
@@ -108,18 +108,32 @@ private static function filter_wporg_app( $app ) {
 
 		if ( self::$plugin_status_adapter->is_plugin_installed( $app['file_path'] ) ) {
 			if ( current_user_can( 'activate_plugins' ) ) {
-				$app['action_label'] = 'Activate';
+<<<<<<< HEAD
```
Comment (hein-obox) [inline modules/apps/admin-apps-page.php]: We need to fix these merging errors.
I can help you if necessary.

Please always try to study the PR, and you can ask me questions here about unexpected changes.

File: modules/promotions/admin-menu-items/custom-icons-promotion-item.php [hunk 24]
```diff
@@ -22,7 +22,7 @@ public function get_page_title() {
 	protected function get_promotion_title():string {
```
Comment (hein-obox) [inline modules/promotions/admin-menu-items/custom-icons-promotion-item.php]: We shouldn't have any unrelated changes in our PR. Can you please try to pull the latest elementor/main branch?

File: playwright.config.ts (hunk 8 above)
Comment (hein-obox) [reply playwright.config.ts:1]: Please delete. 
Our PRs need to be clean from unrelated changes, otherwise they won't be accepted by the company.
All our code will have to be merged into the official Elementor branch.

File: tests-examples/demo-todo-app.spec.ts (hunk 9 above)
Comment (hein-obox) [reply tests-examples/demo-todo-app.spec.ts:1]: Please remove.

File: assets/dev/scss/frontend/general/_button.scss (hunk 17 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: Please delete.

File: assets/dev/scss/frontend/general/_button.scss [hunk 25]
```diff
@@ -25,16 +25,29 @@
...
 
 	&-icon {
+
+		i {
+			color: var(--button-icon-color, inherit);
+		    font-size: var(--button-icon-size);
+		}
+
 		svg {
 			width: 1em;
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]:
```suggestion
			width: var(--button-icon-size, 1em);
```

File: assets/dev/scss/frontend/general/_button.scss [hunk 26]
```diff
@@ -25,16 +25,29 @@
 	}
 
 	&-icon {
+
+		i {
+			color: var(--button-icon-color, inherit);
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:30]:
```suggestion
		color: var(--button-icon-color, inherit);
```

File: assets/dev/scss/frontend/general/_button.scss (hunk 25 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: You will need to set a size for the SVG element.

File: assets/dev/scss/frontend/general/_button.scss (hunk 18 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:46]: Do you why we have this element here?

It seems to be a duplicate of the SVG styling.

Can you please comment it out for now?

File: assets/dev/scss/frontend/general/_button.scss (hunk 20 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:47]: Not updated?

File: assets/dev/scss/frontend/general/_button.scss (hunk 21 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: Please delete this code.

File: assets/dev/scss/frontend/_widgets.scss (hunk 15 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/_widgets.scss]: i removed it

File: assets/dev/scss/frontend/general/_button.scss (hunk 16 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss:46]: I changed it

File: assets/dev/scss/frontend/general/_button.scss (hunk 25 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: @Mchris-hub Do we not need to set a custom width?

File: assets/dev/scss/frontend/general/_button.scss [hunk 27]
```diff
@@ -25,16 +25,25 @@
...
+		}
+
 		svg {
 			width: 1em;
 			height: auto;
+			fill: var(--button-icon-color, inherit);
+			
 		}
-
+
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Remove tab:

```suggestion

```

File: assets/dev/scss/frontend/general/_button.scss (hunk 18 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:42]: Could we delete this? Do we need this?

File: assets/dev/scss/frontend/general/_button.scss (hunk 20 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:43]: No feedback?

File: assets/dev/scss/frontend/general/_button.scss [hunk 28]
```diff
@@ -25,16 +25,25 @@
...
+			
 		}
-
+			
 		.e-font-icon-svg {
 			height: 1em;
 		}
 	}
-
+
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Remove empty line:

```suggestion
```

File: includes/widgets/traits/button-trait.php (hunk 13 above)
Comment (hein-obox) [reply includes/widgets/traits/button-trait.php]: Let's leave it for now. We can check it later.

File: assets/dev/scss/frontend/widgets/button.scss (hunk 14 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/button.scss]: Can you make sure that you use the value from the control and the hardcoded value?

File: assets/dev/scss/frontend/general/_button.scss (hunk 19 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: I have set the styling of the size of the icon to the existing styling

File: assets/dev/scss/frontend/general/_button.scss (hunk 21 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: I have nested the code

File: assets/dev/scss/frontend/general/_button.scss (hunk 25 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: I have added your suggestion, and I think it works the same way. I do not really know the consequences of not having a custom width

File: assets/dev/scss/frontend/general/_button.scss [hunk 29]
```diff
@@ -25,16 +25,25 @@
 	}
 
 	&-icon {
+
+		i {
+			color: var(--button-icon-color, inherit);
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:30]:
```suggestion
		color: var(--button-icon-color, inherit);
```

File: assets/dev/scss/frontend/general/_button.scss [hunk 30]
```diff
@@ -25,16 +25,25 @@
...
+		}
+		
 		svg {
-			width: 1em;
+			width: var(--button-icon-size, 1em);
 			height: auto;
+			fill: var(--button-icon-color, inherit);
 		}
-
+
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Remove tabs.

File: assets/dev/scss/frontend/general/_button.scss (hunk 20 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:43]: Not implemented?

File: assets/dev/scss/frontend/general/_button.scss (hunk 29 above)
Comment (Mchris-hub) [inline assets/dev/scss/frontend/general/_button.scss:38]: Screenshot 2024-04-30 191042
I think your Vscode formats your document automatically because there are no tabs on my side

File: assets/dev/scss/frontend/general/_button.scss (hunk 30 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: image
I think your Vscode formats your document automatically because there are no tabs on my side

File: assets/dev/scss/frontend/general/_button.scss [hunk 31]
```diff
@@ -83,9 +92,27 @@
 
 	span {
 		text-decoration: inherit; //fix for conflict with text-decoration & inline-block
+		align-self: center;
+	}
+
+	&:hover {
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:102]: Nice!!!

File: assets/dev/scss/frontend/general/_button.scss [hunk 32]
```diff
@@ -83,9 +92,27 @@
...
+
+		.elementor-button-icon {
+
+			i {
+				color: var(--button-icon-color-hover);
+				transition: all 0.3s;
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: I assume this styling should be set by the control? And not hardcoded like this?

File: assets/dev/scss/frontend/general/_button.scss (hunk 30 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: Our code needs to be completely clean before it can be merged.

File: assets/dev/scss/frontend/general/_button.scss (hunk 32 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: the transition? I just copied the button styling. I inspected the button on the browser and I saw how they styled it and it was hard coded so I just followed their styling method

File: assets/dev/scss/frontend/general/_button.scss (hunk 30 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: @Mchris-hub Not fixed yet.

File: assets/dev/scss/frontend/general/_button.scss (hunk 18 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:41]: @Mchris-hub No feedback?

File: assets/dev/scss/frontend/general/_button.scss (hunk 28 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: @Mchris-hub No feedback?

File: assets/dev/scss/frontend/general/_button.scss (hunk 30 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: I have cleaned the code and pushed it

File: assets/dev/scss/frontend/widgets/button.scss (hunk 14 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/widgets/button.scss]: when the user hovers over a button there is a delay, I inspected the page and I found there is a hardcoded transition value for the the button so I just copied the styling and applied it to the icon

File: assets/dev/scss/frontend/general/_button.scss (hunk 30 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: @hein-obox like you can see there are not any empty spaces and tabs so I am a little bit confused

File: assets/dev/scss/frontend/general/_button.scss (hunk 32 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: @Mchris-hub I will need feedback.

File: assets/dev/scss/frontend/general/_button.scss (hunk 28 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: @Mchris-hub There are two empty lines. Only one is allowed.

File: assets/dev/scss/frontend/widgets/button.scss (hunk 14 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/button.scss]: @Mchris-hub I will need feedback here.

File: assets/dev/scss/frontend/general/_button.scss (hunk 32 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: I do not understand the nature of the feedback you are expecting, I will say that  it is hard coded for the button too if it is what you want to know

File: assets/dev/scss/frontend/general/_button.scss (hunk 28 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: Screenshot 2024-05-01 153743
do you expect the code to look like this?

File: assets/dev/scss/frontend/general/_button.scss [hunk 33]
```diff
@@ -25,16 +25,23 @@
 	}
 
 	&-icon {
+		i {
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:28]: I am not sure what are you doing, but you don't seem to be using the Github interface to see your changes.

This empty line shouldn't have been removed.

@Mchris-hub

File: assets/dev/scss/frontend/general/_button.scss (hunk 30 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: @Mchris-hub One empty line has been removed, but new tabs have been introduced. This needs to be fixed.

Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: @hein-obox I want to know when you say tabs do you mean that you want the closing semi-colons of each selector to align?

File: assets/dev/scss/frontend/general/_button.scss (hunk 28 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: I have removed the empty lines

File: includes/widgets/traits/button-trait.php (hunk 13 above)
Comment (Mchris-hub) [reply includes/widgets/traits/button-trait.php]: okay

File: assets/dev/scss/frontend/general/_button.scss [hunk 34]
```diff
@@ -25,16 +25,24 @@
...
+			color: var(--button-icon-color, inherit);
+		    font-size: var(--button-icon-size);
+		}
+		
 		svg {
-			width: 1em;
 			height: auto;
+			width: var(--button-icon-size, 1em);
+			fill: var(--button-icon-color, inherit);
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:37]: Maybe we should introduce this CSS variable for the title/text: `--button-icon-color`.

```suggestion
			fill: var(--button-icon-color, --button-title-color);
```

File: assets/dev/scss/frontend/general/_button.scss [hunk 35]
```diff
@@ -83,19 +91,32 @@
...
+				fill: var(--button-icon-color-hover);
+				transition: all 0.3s;
+			}
+		}
 	}
 }
 
 .elementor-element.elementor-button- {
 	@include button-types( '.elementor-button' );
-
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Please restore the original version.

File: assets/dev/scss/frontend/general/_button.scss [hunk 36]
```diff
@@ -83,19 +91,32 @@
...
 }
 
 .elementor-widget-button {
 	.elementor-button {
 		.elementor-button- {
-
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Please restore the original version.

File: assets/dev/scss/frontend/general/_button.scss [hunk 37]
```diff
@@ -83,19 +91,32 @@
...
 			@include button-types( '' );
 		}
 	}
-}
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Please restore the original version. I think we need to remove one line at the end.

File: assets/dev/scss/frontend/general/_button.scss (hunk 34 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:37]: My IDE is giving me an error about this 'inherit' styling value.
Can we try to get rid of it?

File: assets/dev/scss/frontend/general/_button.scss [hunk 38]
```diff
@@ -83,19 +91,32 @@
 
 	span {
 		text-decoration: inherit; //fix for conflict with text-decoration & inline-block
+		align-self: center;
+	}
+
+	&:hover {
+		.elementor-button-icon {
+			i {
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss:106]: Add empty line:

```suggestion

			i {
```

File: assets/dev/scss/frontend/general/_button.scss (hunk 35 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss:91]: I added a line at the end

File: assets/dev/scss/frontend/general/_button.scss (hunk 37 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss:101]: I removed the line at the end

File: assets/dev/scss/frontend/general/_button.scss (hunk 36 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss:97]: I added the line like the original version

File: assets/dev/scss/frontend/general/_button.scss (hunk 34 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss:37]: where is the value for that variable supposed to come?

**Review by Mchris-hub — COMMENTED** [review]
I have added come comments

File: assets/dev/scss/frontend/general/_button.scss (hunk 36 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: You added tabs.

Please check your work on GitHub.

File: assets/dev/scss/frontend/general/_button.scss (hunk 38 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss:106]: I added an empty line

File: assets/dev/scss/frontend/general/_button.scss (hunk 36 above)
Comment (Mchris-hub) [reply assets/dev/scss/frontend/general/_button.scss]: I removed those tabs

File: assets/dev/scss/frontend/general/_button.scss (hunk 32 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss]: The transition duration is added dynamically with this control:

My preferred approach is that you introduce a CSS variable inside this control and replace the existing transition styling with a CSS variable based approach.

Screenshot 2024-05-02 at 13 33 09

File: assets/dev/scss/frontend/general/_button.scss (hunk 18 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:41]: Interesting. 
Can you see what this styling is used for exactly?

File: assets/dev/scss/frontend/general/_button.scss [hunk 39]
```diff
@@ -25,13 +25,21 @@
...
 		svg {
-			width: 1em;
 			height: auto;
+			width: var(--button-icon-size, 1em);
+			fill: var(--button-icon-color, inherit);
 		}
 
 		.e-font-icon-svg {
 			height: 1em;
+			font-size: var(--button-icon-size);
```
Comment (hein-obox) [inline assets/dev/scss/frontend/general/_button.scss]: Do we need this line here?
Is it working?

What happens when you comment it out?

File: assets/dev/scss/frontend/general/_button.scss (hunk 20 above)
[the remark this answers is not in the document]
Comment (hein-obox) [reply assets/dev/scss/frontend/general/_button.scss:43]: I think you are commenting on old comment of mine.
